C error instead of syntax error on incorrect use of << operator

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 3164e29.67c6f24
**OS:** linux, KDE neon User Edition 5.20

So, I'm very new to vlang and I'm still learning it, I eventually tried to use the << operator to append numbers to an array.
Instead of a successful compile or an syntax error I got a C error instead.

**What did you do?**
Tried to append multiple values to array incorrectly with:

        fn main() {

                mut array := [1, 2, 3]

                array << 4, 5

                println(array.str())

        }

Instead of:
`array << [4, 5]`

**What did you expect to see?**
Syntax error
 
**What did you see instead?**

        ==================
        /tmp/v/a.14911027766088884744.tmp.c: In function ‘main__main’:
        /tmp/v/a.14911027766088884744.tmp.c:8505:40: error: void value not ignored as it ought to be
         8505 |  (multi_return_void_int_literal){.arg0=array_push(&array, _MOV((int[]){ 4 })),.arg1=5};
              |                                        ^~~~~~~~~~
        ...
        ==================

--- a/vlib/v/parser/parser.v
+++ b/vlib/v/parser/parser.v
@@ -1020,13 +1020,16 @@ fn (mut p Parser) parse_multi_expr(is_top_level bool) ast.Stmt {
 	if p.tok.kind in [.assign, .decl_assign] || p.tok.kind.is_assign() {
 		return p.partial_assign_stmt(left, left_comments)
 	} else if !p.pref.translated
-		&& tok.kind !in [.key_if, .key_match, .key_lock, .key_rlock, .key_select]
-		&& left0 !is ast.CallExpr && (is_top_level || p.tok.kind != .rcbr)
-		&& left0 !is ast.PostfixExpr && !(left0 is ast.InfixExpr
-		&& (left0 as ast.InfixExpr).op in [.left_shift, .arrow]) && left0 !is ast.ComptimeCall
-		&& left0 !is ast.SelectorExpr {
-		p.error_with_pos('expression evaluated but not used', left0.position())
-		return ast.Stmt{}
+		&& tok.kind !in [.key_if, .key_match, .key_lock, .key_rlock, .key_select] {
+		for node in left {
+			if node !is ast.CallExpr && (is_top_level || p.tok.kind != .rcbr)
+				&& node !is ast.PostfixExpr && !(node is ast.InfixExpr
+				&& (node as ast.InfixExpr).op in [.left_shift, .arrow]) && node !is ast.ComptimeCall
+				&& node !is ast.SelectorExpr {
+				p.error_with_pos('expression evaluated but not used', node.position())
+				return ast.Stmt{}
+			}
+		}
 	}
 	pos.update_last_line(p.prev_tok.line_nr)
 	if left.len == 1 {

--- a/vlib/v/parser/tests/expr_evaluated_but_not_used_d.out
+++ b/vlib/v/parser/tests/expr_evaluated_but_not_used_d.out
@@ -1,7 +1,6 @@
-vlib/v/parser/tests/expr_evaluated_but_not_used_d.vv:4:5: error: expression evaluated but not used
+vlib/v/parser/tests/expr_evaluated_but_not_used_d.vv:4:19: error: expression evaluated but not used
     2 |     a := 1
     3 |     b := 2
-    4 |     a+b
-      |     ~~~
-    5 |     println(a*b)
-    6 | }
+    4 |     println(a*b), a+b
+      |                   ~~~
+    5 | }

--- a/vlib/v/parser/tests/expr_evaluated_but_not_used_d.vv
+++ b/vlib/v/parser/tests/expr_evaluated_but_not_used_d.vv
@@ -1,6 +1,5 @@
 fn main() {
     a := 1
     b := 2
-    a+b
-    println(a*b)
+    println(a*b), a+b
 }

--- a/vlib/v/parser/tests/expr_evaluated_but_not_used_e.out
+++ b/vlib/v/parser/tests/expr_evaluated_but_not_used_e.out
@@ -1,0 +1,7 @@
+vlib/v/parser/tests/expr_evaluated_but_not_used_e.vv:3:14: error: expression evaluated but not used
+    1 | fn main() {
+    2 |     mut array := [1, 2, 3]
+    3 |     array << 4, 5
+      |                 ^
+    4 |     println(array)
+    5 | }

--- a/vlib/v/parser/tests/expr_evaluated_but_not_used_e.vv
+++ b/vlib/v/parser/tests/expr_evaluated_but_not_used_e.vv
@@ -1,0 +1,5 @@
+fn main() {
+	mut array := [1, 2, 3]
+	array << 4, 5
+	println(array)
+}

